pkg/helpers/helpers.go

```diff
@@ -225,3 +246,66 @@ func verifySubscriptionDeleted(ctx context.Context, subscriptionID string, conne
 	connection.Logger().Info(ctx, "Subscription '%s' deleted successfully", subscriptionID)
 	return nil
 }
+
+func DeleteService(ctx context.Context, id string, connection *sdk.Connection) {
+	connection.Logger().Info(ctx, "Deleting service '%s'", id)
+	// Send the request to delete the service
+	response, err := connection.Delete().
+		Path(ServiceEndpoint + id).
+		Send()
+	if err != nil {
+		connection.Logger().Error(ctx, "Failed to delete service '%s', got error: %v", id, err)
+		markFailedServiceCleanup(id)
+	} else if response.Status() != 201 {
```

Comment:
Do we get a non-201 status AND an err ? If so we would never hit this else if as the initial err != nil would catch the error first.

Reply:
According to their swagger doc on the endpoint, they should return in case of success `201`.
In case there was any random error or auth error, so status code `400 or above` they return an error and will go into the first part.
If we get any response that is not an error state, say `200 to 399`, we don't get an error, but we are expecting a `201` so to double check that is the if there, in case they change something in their code we are able to catch it.